app/src/org/commcare/utils/FormUploadUtil.java

```diff
@@ -399,4 +415,29 @@ private static boolean isAudioVisualMimeType(String filename) {
 
         return false;
     }
+
+    public static String parseProcessingFailureResponse(InputStream responseStream) {
+        try {
+            KXmlParser baseParser = ElementParser.instantiateParser(responseStream);
+            ElementParser<String> responseParser = new ElementParser<String>(baseParser) {
+                @Override
+                public String parse() throws InvalidStructureException, IOException,
+                        XmlPullParserException, UnfullfilledRequirementsException {
+                    checkNode("OpenRosaResponse");
+                    nextTag("message");
+                    String natureOfResponse = parser.getAttributeValue(null, "nature");
+                    if ("processing_failure".equals(natureOfResponse)) {
+                        return parser.nextText();
+                    }
+                    return "";
+                }
+            };
+            return responseParser.parse();
+        } catch (IOException | InvalidStructureException | XmlPullParserException |
+                UnfullfilledRequirementsException e) {
+            Log.e(TAG, "Error while parsing form upload response");
```

Comment:
I think we should use Logger.exception() for this so we can track why we wren't able to process hte response.

Also: I thought we explicitly required that the response both be a 422 *and* be a valid openrosa response payload in order for us to quarantine the result, so swallowing this silently feels dangerous to me. am I misremembering?